modules/swift/examples/code_snippets/SampleCodeTest.swift

```diff
@@ -1209,2824 +1208,1603 @@ class SampleCodeTest {
     }
-
-// tag::xctListener-auth-tls-ca-cert-full[]
+    
+    // tag::xctListener-auth-tls-ca-cert-full[]
     func testServerCertVerificationModeCACert() throws {
-        if !self.keyChainAccessAllowed {
-            return
-        }
-
         // Listener:
-// tag::xctListener-auth-tls-ca-cert[]
-        let listener = try listen(tls: true)
-        XCTAssertNotNil(listener.tlsIdentity)
-        XCTAssertEqual(listener.tlsIdentity!.certs.count, 1)
-
-        // Replicator - Success:
-        self.ignoreException {
-            let serverCert = listener.tlsIdentity!.certs[0]
-            self.run(target: listener.localURLEndpoint, type: .pushAndPull, continuous: false,
-                     serverCertVerifyMode: false, serverCert: serverCert)
-        }
-// end::xctListener-auth-tls-ca-cert[]
-
-        // Replicator - TLS Error:
-        self.ignoreException {
-            self.run(target: listener.localURLEndpoint, type: .pushAndPull, continuous: false,
-                     serverCertVerifyMode: false, serverCert: nil, expectedError: CBLErrorTLSCertUnknownRoot)
-        }
-
-        // Cleanup
-        try stopListen()
-    }
-
-    func testPort() throws {
-        if !self.keyChainAccessAllowed {
-            return
-        }
-
-        let config = URLEndpointListenerConfiguration(database: self.oDB)
-        config.port = wsPort
+        // tag::xctListener-auth-tls-ca-cert[]
+        let rootCertPath = Bundle.main.path(forResource: "identity/certs", ofType: "p12")
+        let rootCertData = try NSData(contentsOfFile: rootCertPath!, options: []) as Data
+        var config = URLEndpointListenerConfiguration(database: otherDB)
+        let identity = try TLSIdentity.importIdentity(withData: rootCertData,
+                                                      password: "123",
+                                                      label: "couchbaselite-server-cert-label")
+        config.tlsIdentity = identity
         self.listener = URLEndpointListener(config: config)
-        XCTAssertNil(self.listener!.port)
-
-        // Start:
-        try self.listener!.start()
-        XCTAssertEqual(self.listener!.port, wsPort)
-
-        try stopListen()
-        XCTAssertNil(self.listener!.port)
-    }
-
-    func testEmptyPort() throws {
-        if !self.keyChainAccessAllowed {
-            return
-        }
-
-        let config = URLEndpointListenerConfiguration(database: self.oDB)
+        try self.listener.start()
+        
+        print(self.listener.tlsIdentity!.certs) // `self.listener.tlsIdentity` TLSIdentity with CA
+        
+        // Replicator
+        let target = URLEndpoint(url: URL(string: "wss://localhost:4985/otherDB")!)
+        var replicatorConfig = ReplicatorConfiguration(database: database, target: target)
+        replicatorConfig.replicatorType = .pushAndPull
+        replicatorConfig.continuous = false
+        replicatorConfig.pinnedServerCertificate = self.listener.tlsIdentity!.certs[0]
```

Comment:
Same as the comment above. The cert needs to be sent from the listener app to the client. Please add some note.

Reply:
not used!